server/server/src/file_service.rs

```diff
@@ -202,6 +221,23 @@ pub async fn change_doc(
         )?
         .to_lazy();
 
+        let old_usage = get_usage(&tree, db.sizes.data())
+            .unwrap_or_default()
+            .iter()
+            .map(|f| f.size_bytes)
+            .sum::<u64>();
+        let new_usage = old_usage + request.new_content.value.len() as u64;
```

Comment:
this is not correct, you're double counting the old size + the new size.

Reply:
imo you should probably, just here, calculate the 
`old_size = sizes.get(id) + METADATA_FEE`, and `new_size = request.new.len() + METADATA_FEE`

don't want to incur the cost of sizing all metadatas for no reason.

Reply:
i still believe calculating the cost of all metadata is necessary. even if you're increasing the size of one document, you have to be aware of the total current usage to determine if you're going over the total usage cap.